src/layers_oneDNN/ConvLayer.cpp

```diff
@@ -0,0 +1,522 @@
+
+  size_t kernel_out_channels, kernel_height, kernel_width;
+
+  if (use_legacy_ ||
+      (kernel_shape.dims() == 4 && kernel_shape[3] > kernel_shape[2])) {
+    kernel_height = kernel_shape[0];
+    kernel_width = kernel_shape[1];
+    kernel_out_channels = kernel_shape[3];
+  } else {
+    kernel_out_channels = kernel_shape[0];
+    kernel_height = kernel_shape[2];
+    kernel_width = kernel_shape[3];
+  }
+
+  size_t batch_size = input_shape[0];
+  size_t input_height = input_shape[2];
+  size_t input_width = input_shape[3];
+
+  size_t effective_kernel_height = (kernel_height - 1) * dilations_ + 1;
+  size_t effective_kernel_width = (kernel_width - 1) * dilations_ + 1;
+
+  size_t output_height =
+      (input_height + 2 * pads_ - effective_kernel_height) / stride_ + 1;
+  size_t output_width =
+      (input_width + 2 * pads_ - effective_kernel_width) / stride_ + 1;
+
+  return Shape({batch_size, kernel_out_channels, output_height, output_width});
+}
+
+void ConvLayerOneDnn::initialize_special_conv(const Shape& input_shape,
+                                              Type data_type) {
+  try {
+    engine_ = std::make_unique<dnnl::engine>(dnnl::engine::kind::cpu, 0);
+    stream_ = std::make_unique<dnnl::stream>(*engine_);
+
+    dnnl::memory::dims src_dims = shape_to_dims(input_shape);
+
+    Shape output_shape = get_output_shape(input_shape);
+    dnnl::memory::dims dst_dims = shape_to_dims(output_shape);
+
+    for (auto d : dst_dims) std::cout << d << " ";
+    std::cout << std::endl;
+
+    dnnl::memory::dims strides = {static_cast<dnnl::memory::dim>(stride_),
+                                  static_cast<dnnl::memory::dim>(stride_)};
+    dnnl::memory::dims padding = {static_cast<dnnl::memory::dim>(pads_),
+                                  static_cast<dnnl::memory::dim>(pads_)};
+    dnnl::memory::dims dilation = {
+        static_cast<dnnl::memory::dim>(dilations_ - 1),
+        static_cast<dnnl::memory::dim>(dilations_ - 1)};
+
+    dnnl::memory::data_type dt = dnnl::memory::data_type::f32;
+
+    auto src_md =
+        dnnl::memory::desc(src_dims, dt, dnnl::memory::format_tag::nchw);
+    auto dst_md =
+        dnnl::memory::desc(dst_dims, dt, dnnl::memory::format_tag::nchw);
+
+    const auto& k_shape = kernel_.get_shape();
+
+    dnnl::memory::dims weights_dims = {
+        static_cast<dnnl::memory::dim>(k_shape[3]),
+        static_cast<dnnl::memory::dim>(k_shape[2]),
+        static_cast<dnnl::memory::dim>(k_shape[0]),
+        static_cast<dnnl::memory::dim>(k_shape[1])};
+
+    auto weights_md =
+        dnnl::memory::desc(weights_dims, dt, dnnl::memory::format_tag::oihw);
+
+    dnnl::convolution_forward::primitive_desc conv_pd =
+        dnnl::convolution_forward::primitive_desc(
+            *engine_, dnnl::prop_kind::forward_inference,
+            dnnl::algorithm::convolution_direct, src_md, weights_md,
+            dnnl::memory::desc(), dst_md, strides, dilation, padding, padding);
```

Comment:
Should we check is there is bias or not? I'm not sure empty desc is a good approach here.